src/RepairManager/src/rule.py

```diff
@@ -0,0 +1,421 @@
+#!/usr/bin/env python3
+
+import logging
+import os
+
+from util import State, RestUtil, PrometheusUtil, walk_json
+
+logger = logging.getLogger(__name__)
+
+
+def override(func):
+    """Return the func itself. Serves as an override decorator."""
+    return func
+
+
+CHROOT_HOST_FS = "chroot /host-fs "
+
+
+def exec_command(commands):
+    """Execute command on host filesystem
+
+    Args:
+        commands: A single command or a list of commands to execute.
+
+    Returns:
+        None
+    """
+    if not isinstance(commands, list):
+        commands = [commands]
+    for command in commands:
+        command = CHROOT_HOST_FS + command
+        logger.info("Executing command: %s", command)
+        os.system(command)
```

Comment:
Maybe use `Popen` to capture stdout and stderr and log. Use os.system directly will put them to current process stdout and stderr, which might lost when docker restart. And Popen also provide timeout.

Also, maybe the function name should be `exec_commands`. Should make it accept list only. And command should better be in form of list, the string format will invoke a sh to interpret the command.

I think it's better to define two functions:

* exec_command
* exec_commands

The first accept a list, the second accept a list of list.

With above change:
* `exec_command("reboot -f")` -> `exec_command(["reboot", "-f"])`
* `exec_command(["nvsm dump health", "sync", "reboot -f"])` -> `exec_commands([["nvsm", "dump", "health"], ["sync"], ["reboot", "-f"]])`

Reply:
Different command may have different timeout. Implemented `exec_cmd(command)` where `command` is a list.